### ISSUE TITLE
flake8 integration fails to ignore things that should be ignored

### ISSUE THREAD

**Issue by dimbleby** [palantir/python-language-server#676]
eg if I have 

```
[flake8]
ignore = W503,E203
```

in `~/.config/flake8`, those errors are reported by `pyls` anyway.

I think what's going on is that `flake8` only filters results when passing them to its formatter, round about [here](https://github.com/PyCQA/flake8/blob/e0612597094234cd7acaad3d64b6c7071cdf1831/src/flake8/style_guide.py#L573). But `pyls`'s `flake8` plugin looks at the unfiltered results.

**Comment by youben11** [comment]
which plugins are enabled in your local setup? flake8 plugin is disabled by default and should be enabled as well as flake8 configuration file parsing so that you have the expected results. Also if you are using multiple linter plugins than the report might be from another linter other than flake8, if you wanna use flake8 only then make sure it's the only one enabled.

**Comment by dimbleby** [comment]
Yes, it is the only one that is enabled.  The source of the warnings is definitely the flake8 plugin, I looked at the message that the language server is sending.

**Comment by youben11** [comment]
what about the flake8 configuration files parsing? what's the value for "pyls.configurationSources"? it should include "flake8" so that flake8 files are parsed.

**Comment by dimbleby** [comment]
Yes, flake8 is among the configuration sources.  I checked the logs [here](https://github.com/palantir/python-language-server/blob/8db7654f9c3786f579416ec42e6a6625003ee43a/pyls/plugins/flake8_lint.py#L19), and find that we have `{'ignore': ['W503', 'E203'], max_line_length': 88}`.

Did you try reproducing this yourself?  Of course I can be wrong, but I really think that the problem is what I said:

> flake8 only filters results when passing them to its formatter, round about here. But pyls's flake8 plugin looks at the unfiltered results.

Cheers.

**Comment by youben11** [comment]
I made sure the setup was as expected. I now reproduced it and got your point. I implemented the linting here #656 using the internal API which doesn't seem to filter using the ignore parameter when generating the report, instead the filtering will occur later (maybe while printing out results to the user). I didn't want to do the linting via calling the flake8 utility but I think that I will have to, this should fix this issue.
